src/run/run.c

```diff
@@ -117,8 +112,22 @@ static void help(void) {
                , program_invocation_short_name);
 }
 
-static bool with_timer(void) {
-        return arg_on_active || arg_on_boot || arg_on_startup || arg_on_unit_active || arg_on_unit_inactive || arg_on_calendar;
+static int add_timer_property(const char *name, const char *val) {
+        _cleanup_free_ char *p = NULL;
+
+        assert(name);
+        assert(val);
+
+        p = strjoin(name, "=", val);
+        if (!p)
+                return log_oom();
+
+        if (strv_consume(&arg_timer_property, p) < 0)
+                return log_oom();
+
+        p = NULL;
+
+        return 0;
```

Comment:
hmm, I am not sure I like this approach I must say, as the error messages will be confusing if things are done that way, as they'd claim property assignments that were made that aren't actually made... I think it would be nicer to just add a struct for this: 

```
typedef struct TimerEvent {
        const char *type;
        usec_t t;
} TimerEvent;

static TimerEvent *arg_timer_events = NULL;
static size_t arg_n_timer_events = 0;
```

With a bit of GREEDY_REALLOC() the code for this shouldn't be much longer than the above, but the error messages would be generated right-away and we wouldn't synthesize property strings internally that we then throw away after parsing them again...

Reply:
Sorry, I do not catch your point... I just intend that this makes `--on-active=` and friends be just shortcuts of `--timer-property=OnActiveSec=` or something.

Reply:
hmm, ok, i figure you could see it that way. I'd still prefer if we'd not parse this stuff twice though... i.e. convert things from strings to our internal representation once, and then only use that...

Reply:
Yeah, I agree with you about that point. But I do not have good idea to deal with.